### PR TITLE
Create a cryptolib.Signer interface to Cloud KMS

### PR DESCRIPTION
**Pull request by nenaddedic**
I had to update vendor/ because the existing kms client libs didn't have any of the asymmetric crypto methods, which are required for signing.

### CODE REVIEW

**Comment by ooq** [comment]
FYI, #518 is merged. Updating with master should pass this PR on the GCB test.
Let's add an integration test for KMS signing as well. As part of it, we probably need to add KMS access to GCB service account.

File: pkg/kritis/signer/cloud_kms.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+func (s kmsSigner) CreateAttestation(payload []byte) (*cryptolib.Attestation, error) {
+	var digest hash.Hash
+	switch s.digestAlg {
+	case SHA256:
+		digest = sha256.New()
+	case SHA384:
+		digest = sha512.New384()
+	case SHA512:
+		digest = sha512.New()
```
Comment (ooq) [inline pkg/kritis/signer/cloud_kms.go:67]: Error handling for non-supported cases?

File: hack/dep.sh [hunk 2]
```diff
@@ -53,11 +53,8 @@ files=( $(validate_diff --name-only -- 'Gopkg.toml' 'Gopkg.lock' 'vendor/' || tr
 unset IFS
 
 if [ ${#files[@]} -gt 0 ]; then
-	if ! [ -x "$(command -v dep)" ]; then
-		install_dep
```
Comment (ooq) [inline hack/dep.sh:57]: This shouldn't be removed?

Comment (nenaddedic) [reply hack/dep.sh:57]: Oops, done.

File: pkg/kritis/signer/cloud_kms.go (hunk 1 above)
Comment (nenaddedic) [reply pkg/kritis/signer/cloud_kms.go:67]: Done.

**Pull request merged**